src/native/fs/fs_napi.cpp

```diff
@@ -230,7 +235,11 @@ parse_open_flags(std::string flags)
     return bits;
 }
 
-const static std::vector<std::string> GPFS_XATTRS{ GPFS_ENCRYPTION_XATTR_NAME };
+const static std::vector<std::string> GPFS_XATTRS{
+	GPFS_ENCRYPTION_XATTR_NAME,
+	GPFS_DMAPI_XATTR_TAPE_INDICATOR,
+	GPFS_DMAPI_XATTR_TAPE_PREMIG,
+};
```

Comment:
this list is going to be used in every stat call... I think we want to optimize it so that it will not include dmapi xattrs when config.NSFS_GLACIER_USE_DMAPI is false. 

For this I would add this to namespace_fs in prepare_fs_context to set fs_context.use_dmapi = config.NSFS_GLACIER_USE_DMAPI. This is also needed in manage_nsfs_glacier in force_gpfs_fs_context ().

Then back here in fs_napi you can store it in FSWorker and pass it to get_fd_gpfs_xattr... a long way... but it would save a these two calls when the config is false which I think is worth the coding effort.

Reply:
Done, PTAL.